kyo-data/shared/src/main/scala/kyo/Chunk.scala

```diff
@@ -328,7 +349,7 @@ sealed abstract class Chunk[A] extends Seq[A] derives CanEqual:
                 case c: Compact[A] =>
                     val l = c.array.length
                     if l > 0 then
-                        System.arraycopy(c.array, dropLeft, array, start, l - dropRight - dropLeft)
+                        Array.copy(c.array, dropLeft, array, start, l - dropRight - dropLeft)
```

Comment:
`Array.copy` is more correct for Scala because it has fallback for dealing with `boxed`/`unboxed` primitive types. It does call `System.arraycopy` when possible.

Reply:
Did you observe issues related to this? The underlying arrays are always boxed in `Chunk`

Reply:
When you need to provide `Chunk[Byte]` to a java method that requires `[]byte`. It'll throw a runtime exception about mismatch between `[]byte` and `[]array`.

Reply:
One work around I used to do is to convert `Chunk` to `List` first, then `.toArray` that `List`. I can do it if you want